pkg/descheduler/descheduler.go

```diff
@@ -416,6 +410,28 @@ func RunDeschedulerStrategies(ctx context.Context, rs *options.DeschedulerServer
 	return nil
 }
 
+func includeProfilePluginsByType(enabledDeschedulePlugins []framework.DeschedulePlugin, enabledBalancePlugins []framework.BalancePlugin, pg framework.Plugin) ([]framework.DeschedulePlugin, []framework.BalancePlugin) {
+	enabledDeschedulePlugins = includeDeschedule(enabledDeschedulePlugins, pg)
+	enabledBalancePlugins = includeBalance(enabledBalancePlugins, pg)
+	return enabledDeschedulePlugins, enabledBalancePlugins
+}
+
+func includeDeschedule(enabledDeschedulePlugins []framework.DeschedulePlugin, pg framework.Plugin) []framework.DeschedulePlugin {
+	switch p := pg.(type) {
```

Comment:
do you need a type switch for these or could you just try casting to `framework.DeschedulePlugin` and only append if it succeeds?

Reply:
did that! :heavy_check_mark: